lgn-worker/src/grpc/mod.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::Result;
+use protobuf::worker_to_gw_request;
+use protobuf::WorkerToGwRequest;
+use protobuf::WorkerToGwResponse;
+use tokio::sync::mpsc;
+use tokio_stream::StreamExt;
+use tonic::metadata::MetadataValue;
+use tonic::Request;
+use tracing::error;
+use tracing::info;
+
+pub mod test;
+
+pub mod protobuf
+{
+    tonic::include_proto!("lagrange");
+}
+const MAX_GRPC_MESSAGE_SIZE_MB: usize = 16;
+
+pub struct GrpcConfig
+{
+    /// Gateway URI
+    pub gateway: String,
+    /// Version of the worker to advertise to the GW
+    pub version: String,
```

Comment:
Should/could this be a [Version](https://docs.rs/semver/1.0.23/semver/struct.Version.html) to move as much parsing downstream as can be?